relayer/chains/icon/provider.go

```diff
@@ -9,25 +9,15 @@ import (
 	"github.com/icon-project/centralized-relay/relayer/kms"
 	"github.com/icon-project/centralized-relay/relayer/provider"
 	providerTypes "github.com/icon-project/centralized-relay/relayer/types"
-	relayerTypes "github.com/icon-project/centralized-relay/relayer/types"
 	"github.com/icon-project/goloop/module"
 	"go.uber.org/zap"
 )
 
 type Config struct {
-	ChainName      string                         `json:"-" yaml:"-"`
-	RPCUrl         string                         `json:"rpc-url" yaml:"rpc-url"`
-	Address        string                         `json:"address" yaml:"address"`
-	StartHeight    uint64                         `json:"start-height" yaml:"start-height"` // would be of highest priority
-	Contracts      relayerTypes.ContractConfigMap `json:"contracts" yaml:"contracts"`
-	NetworkID      int64                          `json:"network-id" yaml:"network-id"`
-	FinalityBlock  uint64                         `json:"finality-block" yaml:"finality-block"`
-	NID            string                         `json:"nid" yaml:"nid"`
-	StepMin        int64                          `json:"step-min" yaml:"step-min"`
-	StepLimit      int64                          `json:"step-limit" yaml:"step-limit"`
-	StepAdjustment int64                          `json:"step-adjustment" yaml:"step-adjustment"`
-	Disabled       bool                           `json:"disabled" yaml:"disabled"`
-	HomeDir        string                         `json:"-" yaml:"-"`
+	provider.CommonConfig `json:",inline" yaml:",inline"`
```

Comment:
What is the incentive to create common Config? I think config is tightly coupled to each chain. Even though the fields in Config are same for chains, it creates a rigidity and is hard to make changes when needed. For example in SUI, we don't have this 
`Contracts     types.ContractConfigMap` field. Contracts fields need to defined in a different way.